spec/models/course_spec.rb

```diff
@@ -94,6 +94,66 @@
         end
       end
 
+      describe '#current_level_threshold' do
+        subject do
+          -> (course, experience_points) { course.current_level_threshold experience_points }
+        end
+
+        context 'when there are no levels created in the course' do
+          let!(:course_without_levels) { create(:course) }
```

Comment:
Possibly then the course should have a transient attribute for the number of levels defined.